lib/num_option.fz

```diff
@@ -67,10 +67,10 @@ is
   #
   public val T
     pre
-      safety: (num_option.this??)
+      safety: num_option.this??
   =>
     num_option.this ? v T => v
-                    | nil => fuzion.std.panic "num_option.val called on nil"
+                    | nil => fuzion.std.panic "You either turned off safety or you discovered a severe compiler bug! This code should be unreachable."
```

Comment:
The new message is helpful, but please keep the original cause `num_option.val called on nil` as part of the message. Otherwise it might be hard to debug if this ever shows up (and it will show up, there will be configurations that favor speed over safety). 
Same for the other messages.

Reply:
@fridis: okay, but we should also really improve our tooling to make this easy to debug.